lib/elastic_apm/instrumenter.rb

```diff
@@ -143,32 +143,29 @@ def start_span(
       action: nil,
       backtrace: nil,
       context: nil,
-      trace_context: nil
+      trace_context: nil,
+      transaction: nil
     )
-      return unless (transaction = current_transaction)
-      return unless transaction.sampled?
-
-      transaction.inc_started_spans!
+      parent_transaction = transaction || current_transaction
+      return unless parent_transaction
+      return unless parent_transaction.sampled?
 
-      if transaction.max_spans_reached?
-        transaction.inc_dropped_spans!
-        return
-      end
+      parent = transaction || current_span || current_transaction
```

Comment:
See my example here, https://github.com/elastic/apm-agent-ruby/pull/571#issuecomment-544682821. The `parent` is the span, but we're also passing a transaction (?) to the async `start_span` calls.

My point is, this line would set the parent to `txn` from the example where as it should be `span 1`?